src/main/java/org/sagebionetworks/bridge/services/PermissionService.java

```diff
@@ -158,6 +172,120 @@ public void deletePermission(String appId, String guid) {
+                        Set<AccessLevel> oldAccessLevels = oldPermissions.get(entityType).get(entityId);
+                        Set<AccessLevel> additionalAccessLevels = new HashSet<>();
+                        for (AccessLevel accessLevel : newPermissionEntityTypeMap.get(entityId)) {
+                            if (!oldAccessLevels.contains(accessLevel)) {
+                                additionalAccessLevels.add(accessLevel);
+                            }
+                        }
+                        Map<String, Set<AccessLevel>> entityTypeMap = additionalPermissions.getOrDefault(entityType, new HashMap<>());
+                        entityTypeMap.put(entityId, additionalAccessLevels);
+                        additionalPermissions.put(entityType, entityTypeMap);
+                    } else {
+                        Map<String, Set<AccessLevel>> entityTypeMap = additionalPermissions.getOrDefault(entityType, new HashMap<>());
+                        entityTypeMap.put(entityId, newPermissionEntityTypeMap.get(entityId));
+                        additionalPermissions.put(entityType, entityTypeMap);
+                    }
+                }
+            } else {
+                // the old permissions do not have any of the new so add them all
+                additionalPermissions.put(entityType, newPermissions.get(entityType));
+            }
+        }
+        
+        for (EntityType entityType : oldPermissions.keySet()) {
+            if (newPermissions.containsKey(entityType)) {
+                Map<String, Set<AccessLevel>> oldPermissionEntityTypeMap = newPermissions.get(entityType);
+                for (String entityId : oldPermissionEntityTypeMap.keySet()) {
+                    if (newPermissions.get(entityType).containsKey(entityId)) {
+                        Set<AccessLevel> newAccessLevels = newPermissions.get(entityType).get(entityId);
+                        Set<AccessLevel> removableAccessLevels = new HashSet<>();
+                        for (AccessLevel accessLevel : oldPermissionEntityTypeMap.get(entityId)) {
+                            if (!newAccessLevels.contains(accessLevel)) {
+                                removableAccessLevels.add(accessLevel);
+                            }
+                        }
+                        Map<String, Set<AccessLevel>> entityTypeMap = removablePermissions.getOrDefault(entityType, new HashMap<>());
+                        entityTypeMap.put(entityId, removableAccessLevels);
+                        removablePermissions.put(entityType, entityTypeMap);
+                    } else {
+                        Map<String, Set<AccessLevel>> entityTypeMap = removablePermissions.getOrDefault(entityType, new HashMap<>());
+                        entityTypeMap.put(entityId, oldPermissionEntityTypeMap.get(entityId));
+                        removablePermissions.put(entityType, entityTypeMap);
+                    }
```

Comment:
Similarly, what would hit this code block?

Reply:
Same as above, can't hit it. Removing.